tests/Doctrine/Tests/Common/Persistence/Mapping/ClassMetadataFactoryTest.php

```diff
@@ -159,6 +159,35 @@ public function testWillIgnoreCacheEntriesThatAreNotMetadataInstances()
 
         self::assertSame($metadata, $this->cmf->getMetadataFor('Foo'));
     }
+
+    public function testFallbackMetadataShouldBeCached()
+    {
+        /** @var Cache|PHPUnit_Framework_MockObject_MockObject $cacheDriver */
+        $cacheDriver = $this->createMock(Cache::class);
+        $cacheDriver->expects($this->once())->method('save');
```

Comment:
`once()` is a static method, so we shouldn't call it dynamically.

```suggestion
        $cacheDriver->expects(self::once())->method('save');
```

Reply:
I'd rather stay with `$this->once()`, even [PHPUnit's docs](https://phpunit.readthedocs.io/en/7.3/test-doubles.html) is using this format. Also I haven't seen `self::once()` until today I think :D

Reply:
@malarzm we're using that in other repos (and even in this class) :+1:

Reply:
This is also a requirement for `phpstan-strict-rules` BTW :grin: